spec/features/favorites/user_sees_favorite_indicator_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe 'As a visitor', type: :feature do
+  it 'I see a favorite indicator in my navigation bar on any page with count of pets in my favorites list' do
+    visit '/'
+
+    within('#site-header') do
+      expect(page).to have_content('Favorites (0)')
+    end
+
+    visit '/pets'
+
+    within('#site-header') do
+      expect(page).to have_content('Favorites (0)')
+    end
+
+    visit '/shelters'
+
+    within('#site-header') do
+      expect(page).to have_content('Favorites (0)')
+    end
+
+    visit '/shelters/new'
+
+    within('#site-header') do
+      expect(page).to have_content('Favorites (0)')
+    end
```

Comment:
Note: Look into conventional way to test this so we do not need to refactor every time a view is added to the application.